distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/BaseKeyStoreCommand.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.settings;
+
+import joptsimple.OptionSet;
+import org.elasticsearch.cli.EnvironmentAwareCommand;
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.env.Environment;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+
+public abstract class BaseKeyStoreCommand extends EnvironmentAwareCommand {
+
+    KeyStoreWrapper keyStore;
+    SecureString keyStorePassword;
+    boolean keyStoreMustExist = false;
+
+    public BaseKeyStoreCommand(String description) {
+        super(description);
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        try {
+            final Path configFile = env.configFile();
+            keyStore = KeyStoreWrapper.load(configFile);
+            if (keyStore == null) {
+                if (keyStoreMustExist) {
+                    throw new UserException(ExitCodes.DATA_ERROR, "Elasticsearch keystore not found at [" +
+                        KeyStoreWrapper.keystorePath(env.configFile()) + "]. Use 'create' command to create one.");
+                } else {
+                    if (terminal.promptYesNo("The elasticsearch keystore does not exist. Do you want to create it?", false) == false) {
+                        terminal.println("Exiting without creating keystore.");
+                        return;
+                    }
+                }
+                // allow the user to decide if an auto-created keystore should be password protected
+                keyStorePassword =
+                    terminal.promptYesNo("Do you want to protect the keystore with a password?", false) == false ?
```

Comment:
This won't work from eg package installation. Can we just always auto create iwthout a password, and allow the user to change through the normal passphrase setting command?